src/s390_crypto.c

```diff
@@ -318,14 +325,114 @@ static void set_switches(int msa)
 			*s390_kdsa_functions[n].enabled = 1;
 }
 
+unsigned int is_device_online(const char *dev)
+{
+	unsigned int ret = 0;
+	FILE *file;
+	char c;
+
+	if ((file = fopen(dev, "r")) == NULL)
+		return 0;
+
+	/* Check if device online: 0 = offline, 1 = online */
+	if ((c = fgetc(file)) == '1')
+		ret = 1;
+
+	fclose(file);
+
+	return ret;
+}
+
+unsigned int get_device_type(const char *dev, char *devtype)
+{
+	unsigned int ret = 0;
+	char *type = NULL;
+	size_t size = 0;
+	FILE *file;
+
+	file = fopen(dev, "r");
+	if (file == NULL)
+		return 0;
+
+	/* Read device type: e.g. 'CEX5C' */
+	if (getline(&type, &size, file) == -1)
+		goto end;
+
+	/* Did we get something like 'CEX5C\n'? */
+	if (strlen(type) != 6 || strstr(type, "CEX") != &type[0])
+		goto end;
+
+	memcpy(devtype, type, 5);
+	ret = 1;
+
+end:
+	free(type);
+	fclose(file);
+
+	return ret;
+}
+
+#define AP_PATH  "/sys/devices/ap"
+#define MAX_DEV_LEN 280
```

Comment:
There is a PATH_MAX define in linux/limits.h.

Reply:
Yes, but it's only 255 bytes long and causes a compiler warning:

> s390_crypto.c:401:34: warning: '%s' directive output may be truncated writing up to 255 bytes into a region of size 239 [-Wformat-truncation=]
>   401 |   snprintf(dev, MAX_DEV_LEN, "%s/%s/type", AP_PATH, direntp->d_name);

The warning disappears when increasing the array to 280 bytes.